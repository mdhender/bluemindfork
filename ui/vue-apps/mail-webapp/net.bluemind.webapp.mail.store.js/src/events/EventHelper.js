// FIXME: move to new store

import { DateComparator } from "@bluemind/date";
import { WeekDay } from "@bluemind/i18n";
import injector from "@bluemind/inject";

export default {
    adapt(event, userName) {
        const main = event.value.main;
        return {
            summary: main.summary,
            organizer: {
                name: main.organizer.commonName,
                mail: main.organizer.mailto
            },
            date: adaptDate(main.dtstart, main.dtend, main.rrule),
            attendees: main.attendees.map(attendee => ({ name: attendee.commonName, mail: attendee.mailto })),
            status: adaptStatus(main.attendees, userName),
            uid: event.uid,
            serverEvent: event
        };
    }
};

function adaptDate(dtstart, dtend, rrule) {
    const vueI18n = injector.getProvider("i18n").get();

    // recurrent event
    if (rrule) {
        return adaptRecurrentEvent(dtstart, dtend, rrule, vueI18n);
    }

    // all day event
    if (dtstart.precision === "Date") {
        return adaptAllDayEvent(dtstart, dtend, vueI18n);
    }

    const startDate = new Date(dtstart.iso8601);
    const endDate = new Date(dtend.iso8601);
    if (DateComparator.isSameDay(startDate, endDate)) {
        // same day, different hours
        return vueI18n.t("common.duration.sameday", {
            date: vueI18n.d(startDate, "full_date_long"),
            startDate: vueI18n.d(startDate, "short_time"),
            endDate: vueI18n.d(endDate, "short_time")
        });
    } else {
        // different days, different hours
        return vueI18n.t("common.duration.weekday", {
            startDate: vueI18n.d(startDate, "full_date_time_long"),
            endDate: vueI18n.d(endDate, "full_date_time_long")
        });
    }
}

function adaptRecurrentEvent(dtstart, dtend, rrule, vueI18n) {
    let res = "";
    const startDate = new Date(dtstart.iso8601);
    const endDate = new Date(dtend.iso8601);
    const startTime = vueI18n.d(startDate, "short_time");
    const endTime = vueI18n.d(endDate, "short_time");

    if (rrule.frequency === "WEEKLY") {
        let days = rrule.byDay.map(o => " " + vueI18n.t("common.day_prefix") + " " + WeekDay.compute(o.day));
        let displayedDays = "";
        if (rrule.byDay.length > 1) {
            const lastDay = days.pop();
            displayedDays += days.join(",") + " " + vueI18n.t("common.and") + " " + lastDay;
        } else {
            displayedDays += days.join(",");
        }
        res =
            dtstart.precision === "DateTime"
                ? vueI18n.tc("common.every_week.with_time", rrule.interval, {
                      count: rrule.interval === 1 ? 0 : rrule.interval,
                      days: displayedDays,
                      startTime,
                      endTime
                  })
                : vueI18n.tc("common.every_week", rrule.interval, {
                      count: rrule.interval === 1 ? 0 : rrule.interval,
                      days: displayedDays
                  });
    } else if (rrule.frequency === "DAILY") {
        res =
            dtstart.precision === "DateTime"
                ? vueI18n.t("common.every_day.with_time", { startTime, endTime })
                : vueI18n.t("common.every_day");
    } else if (rrule.frequency === "MONTHLY") {
        if (rrule.byDay.length === 1) {
            const dayOfMonth = rrule.byDay[0];
            const numberSuffix = vueI18n.tc("common.number_suffix", dayOfMonth.offset, { number: dayOfMonth.offset });
            res =
                dtstart.precision === "DateTime"
                    ? vueI18n.t("common.every_month.same_day.with_time", {
                          number: numberSuffix,
                          day: WeekDay.compute(dayOfMonth.day),
                          startTime,
                          endTime
                      })
                    : vueI18n.t("common.every_month.same_day", {
                          number: numberSuffix,
                          day: WeekDay.compute(dayOfMonth.day)
                      });
        } else {
            res =
                dtstart.precision === "DateTime"
                    ? vueI18n.t("common.every_month.same_date.with_time", {
                          date: startDate.getDate(),
                          startTime,
                          endTime
                      })
                    : vueI18n.t("common.every_month.same_date", { date: startDate.getDate() });
        }
    } else if (rrule.frequency === "YEARLY") {
        res =
            dtstart.precision === "DateTime"
                ? vueI18n.t("common.every_year.day_month.with_time", {
                      dayMonth: vueI18n.d(startDate, "day_month"),
                      startTime,
                      endTime
                  })
                : vueI18n.t("common.every_year.day_month", { dayMonth: vueI18n.d(startDate, "day_month") });
    }
    return res;
}

function adaptAllDayEvent(dtstart, dtend, vueI18n) {
    const startDate = new Date(dtstart.iso8601);
    startDate.setHours(0, 0, 0, 0);

    let endDate = new Date(dtend.iso8601);
    endDate.setHours(0, 0, 0, 0);
    endDate = new Date(endDate.getTime() - 1);

    if (DateComparator.isSameDay(startDate, endDate)) {
        return vueI18n.d(startDate, "full_date_long");
    }
    return vueI18n.t("common.duration.weekday", {
        startDate: vueI18n.d(startDate, "full_date_time_long"),
        endDate: vueI18n.d(endDate, "full_date_time_long")
    });
}

/**
 * return null if user is not in attendees
 */
function adaptStatus(attendees, userName) {
    const userSession = injector.getProvider("UserSession").get();
    const currentAddress = userName + "@" + userSession.domain;
    const user = attendees.find(a => a.mailto === currentAddress);
    return user ? user.partStatus : null;
}
