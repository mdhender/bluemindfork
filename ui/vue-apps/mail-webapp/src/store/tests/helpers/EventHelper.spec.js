import allDay from "../data/events/allDay";
import basic from "../data/events/basic";
import recurrentEveryDay from "../data/events/recurrentEveryDay";
import recurrentOnceAWeek from "../data/events/recurrentOnceAWeek";
import recurrentOnceAWeekAllDay from "../data/events/recurrentOnceAWeekAllDay";
import recurrentTwiceAMonth from "../data/events/recurrentTwiceAMonth";

import EventHelper from "../../helpers/EventHelper";

import { InjectI18NDependency } from "@bluemind/test-utils";
InjectI18NDependency.registerCommonL10N();

describe("EventHelper adapter", () => {
    const userUid = "B2CBEEFD-147C-451A-9229-1B6C9697D202";

    test("adapt events match snapshots", () => {
        const events = [
            allDay,
            basic,
            recurrentEveryDay,
            recurrentOnceAWeek,
            recurrentOnceAWeekAllDay,
            recurrentTwiceAMonth
        ];
        events.forEach(e => {
            const adapted = EventHelper.adapt(e, userUid);
            expect(adapted).toMatchSnapshot();
        });
    });
});
