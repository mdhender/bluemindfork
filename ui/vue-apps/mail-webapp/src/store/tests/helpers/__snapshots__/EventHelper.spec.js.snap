// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventHelper adapter adapt events match snapshots 1`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "vendredi 14 août 2020",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336484841,
    "createdBy": "system",
    "displayName": "all day event",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5276,
    "uid": "191319a5-07b8-40bc-80b4-aeae35951029",
    "updated": 1597336484841,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "191319a5-07b8-40bc-80b4-aeae35951029",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -25200,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-15",
          "precision": "Date",
          "timezone": null,
        },
        "dtstart": Object {
          "iso8601": "2020-08-14",
          "precision": "Date",
          "timezone": null,
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": null,
        "sequence": 0,
        "status": "Confirmed",
        "summary": "all day event",
        "transparency": "Transparent",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 27,
  },
  "status": "NeedsAction",
  "summary": "all day event",
  "uid": "191319a5-07b8-40bc-80b4-aeae35951029",
  "url": null,
}
`;

exports[`EventHelper adapter adapt events match snapshots 2`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "vendredi 14 août 2020 de 10:00 à 11:00",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336498670,
    "createdBy": "system",
    "displayName": "one day, one hour",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5280,
    "uid": "0f96bbc1-6c50-42ff-b030-8256df315451",
    "updated": 1597336498670,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "0f96bbc1-6c50-42ff-b030-8256df315451",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -900,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-14T13:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "dtstart": Object {
          "iso8601": "2020-08-14T12:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": null,
        "sequence": 0,
        "status": "Confirmed",
        "summary": "one day, one hour",
        "transparency": "Opaque",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 28,
  },
  "status": "NeedsAction",
  "summary": "one day, one hour",
  "uid": "0f96bbc1-6c50-42ff-b030-8256df315451",
  "url": null,
}
`;

exports[`EventHelper adapter adapt events match snapshots 3`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "Tous les jours, de 10:00 à 11:30, à partir du 17 août 2020",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336529234,
    "createdBy": "system",
    "displayName": "recurrent : every day with an hour",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5285,
    "uid": "69bc4819-b9b6-41ce-bc77-3b648974e953",
    "updated": 1597336529234,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "69bc4819-b9b6-41ce-bc77-3b648974e953",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -900,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-17T13:30:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "dtstart": Object {
          "iso8601": "2020-08-17T12:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": Object {
          "byDay": Array [],
          "byHour": Array [],
          "byMinute": Array [],
          "byMonth": Array [],
          "byMonthDay": Array [],
          "bySecond": Array [],
          "bySetPos": Array [],
          "byWeekNo": Array [],
          "byYearDay": Array [],
          "count": null,
          "frequency": "DAILY",
          "interval": 1,
          "until": null,
        },
        "sequence": 0,
        "status": "Confirmed",
        "summary": "recurrent : every day with an hour",
        "transparency": "Opaque",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 29,
  },
  "status": "NeedsAction",
  "summary": "recurrent : every day with an hour",
  "uid": "69bc4819-b9b6-41ce-bc77-3b648974e953",
  "url": null,
}
`;

exports[`EventHelper adapter adapt events match snapshots 4`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "Toutes les semaines,  le vendredi, de 07:00 à 08:00, à partir du 21 août 2020",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336586337,
    "createdBy": "system",
    "displayName": "once every week with hour",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5295,
    "uid": "fecf2e5a-9149-47ec-8768-6aaca6ffa62b",
    "updated": 1597336586337,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "fecf2e5a-9149-47ec-8768-6aaca6ffa62b",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -900,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-21T10:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "dtstart": Object {
          "iso8601": "2020-08-21T09:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": Object {
          "byDay": Array [
            Object {
              "day": "FR",
              "offset": 0,
            },
          ],
          "byHour": Array [],
          "byMinute": Array [],
          "byMonth": Array [],
          "byMonthDay": Array [],
          "bySecond": Array [],
          "bySetPos": Array [],
          "byWeekNo": Array [],
          "byYearDay": Array [],
          "count": null,
          "frequency": "WEEKLY",
          "interval": 1,
          "until": null,
        },
        "sequence": 0,
        "status": "Confirmed",
        "summary": "once every week with hour",
        "transparency": "Opaque",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 31,
  },
  "status": "NeedsAction",
  "summary": "once every week with hour",
  "uid": "fecf2e5a-9149-47ec-8768-6aaca6ffa62b",
  "url": null,
}
`;

exports[`EventHelper adapter adapt events match snapshots 5`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "Toutes les semaines,  le mardi, à partir du 25 août 2020",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336616186,
    "createdBy": "system",
    "displayName": "once every week ALL_DAY",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5300,
    "uid": "05784243-a2d5-4973-866d-e6ba74b3d0a6",
    "updated": 1597336616186,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "05784243-a2d5-4973-866d-e6ba74b3d0a6",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -25200,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-26",
          "precision": "Date",
          "timezone": null,
        },
        "dtstart": Object {
          "iso8601": "2020-08-25",
          "precision": "Date",
          "timezone": null,
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": Object {
          "byDay": Array [
            Object {
              "day": "TU",
              "offset": 0,
            },
          ],
          "byHour": Array [],
          "byMinute": Array [],
          "byMonth": Array [],
          "byMonthDay": Array [],
          "bySecond": Array [],
          "bySetPos": Array [],
          "byWeekNo": Array [],
          "byYearDay": Array [],
          "count": null,
          "frequency": "WEEKLY",
          "interval": 1,
          "until": null,
        },
        "sequence": 0,
        "status": "Confirmed",
        "summary": "once every week ALL_DAY",
        "transparency": "Opaque",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 32,
  },
  "status": "NeedsAction",
  "summary": "once every week ALL_DAY",
  "uid": "05784243-a2d5-4973-866d-e6ba74b3d0a6",
  "url": null,
}
`;

exports[`EventHelper adapter adapt events match snapshots 6`] = `
Object {
  "attendee": Object {
    "commonName": "test",
    "cutype": "Individual",
    "delFrom": null,
    "delTo": null,
    "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
    "internal": true,
    "lang": null,
    "mailto": "test@webmail-test.loc",
    "member": null,
    "partStatus": "NeedsAction",
    "responseComment": null,
    "role": "RequiredParticipant",
    "rsvp": true,
    "sentBy": null,
    "uri": null,
  },
  "attendees": Array [
    Object {
      "cutype": "Individual",
      "mail": "test@webmail-test.loc",
      "name": "test",
      "status": "NeedsAction",
    },
  ],
  "calendarOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "calendarUid": undefined,
  "conference": undefined,
  "counter": undefined,
  "date": "Le 19 tous les 2 mois, de 07:00 à 07:30, à partir de août 2020",
  "isMeeting": false,
  "isWritable": undefined,
  "loading": "LOADED",
  "location": null,
  "mailboxOwner": "B2CBEEFD-147C-451A-9229-1B6C9697D202",
  "organizer": Object {
    "mail": "jdoe@webmail-test.loc",
    "name": "john doe",
  },
  "private": false,
  "recuridIsoDate": undefined,
  "sanitizedDescription": undefined,
  "serverEvent": Object {
    "created": 1597336556578,
    "createdBy": "system",
    "displayName": "twice a month, with an hour",
    "externalId": null,
    "flags": Array [
      "Seen",
    ],
    "internalId": 5290,
    "uid": "41a4e2e2-8eef-449c-9631-812727514880",
    "updated": 1597336556578,
    "updatedBy": "system",
    "value": Object {
      "icsUid": "41a4e2e2-8eef-449c-9631-812727514880",
      "main": Object {
        "alarm": Array [
          Object {
            "action": "Display",
            "description": null,
            "duration": null,
            "repeat": null,
            "summary": null,
            "trigger": -900,
          },
        ],
        "attachments": Array [],
        "attendees": Array [
          Object {
            "commonName": "test",
            "cutype": "Individual",
            "delFrom": null,
            "delTo": null,
            "dir": "bm://webmail-test.loc/users/B2CBEEFD-147C-451A-9229-1B6C9697D202",
            "internal": true,
            "lang": null,
            "mailto": "test@webmail-test.loc",
            "member": null,
            "partStatus": "NeedsAction",
            "responseComment": null,
            "role": "RequiredParticipant",
            "rsvp": true,
            "sentBy": null,
            "uri": null,
          },
        ],
        "categories": Array [],
        "classification": "Public",
        "description": null,
        "draft": false,
        "dtend": Object {
          "iso8601": "2020-08-19T09:30:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "dtstart": Object {
          "iso8601": "2020-08-19T09:00:00.000+02:00",
          "precision": "DateTime",
          "timezone": "Europe/Paris",
        },
        "exdate": null,
        "location": null,
        "organizer": Object {
          "commonName": "john doe",
          "dir": "bm://webmail-test.loc/users/6926E3AD-6CA1-4147-875F-85F080E64AFA",
          "mailto": "jdoe@webmail-test.loc",
          "uri": null,
        },
        "priority": 5,
        "rdate": null,
        "rrule": Object {
          "byDay": Array [],
          "byHour": Array [],
          "byMinute": Array [],
          "byMonth": Array [],
          "byMonthDay": Array [],
          "bySecond": Array [],
          "bySetPos": Array [],
          "byWeekNo": Array [],
          "byYearDay": Array [],
          "count": null,
          "frequency": "MONTHLY",
          "interval": 2,
          "until": null,
        },
        "sequence": 0,
        "status": "Confirmed",
        "summary": "twice a month, with an hour",
        "transparency": "Opaque",
        "url": null,
      },
      "occurrences": Array [],
      "properties": Object {},
    },
    "version": 30,
  },
  "status": "NeedsAction",
  "summary": "twice a month, with an hour",
  "uid": "41a4e2e2-8eef-449c-9631-812727514880",
  "url": null,
}
`;
